### PR TITLE
Some more code cleanup and refactoring

### PR DESCRIPTION
**Pull request by schell244**
## 🍰 Pullrequest
Code cleanup and some fixes based on Clang-Tidy suggestions.
Also resolved some compiler warnings.
No behavioural changes.

### Proof
- None

### Issues
- None

### How2Test
- None

### Todo / Checklist
- [X] None

### CODE REVIEW

File: src/game/World.h [hunk 1]
```diff
@@ -633,8 +633,8 @@ enum RealmType
 
 class SessionPacketSendTask
 {
-    SessionPacketSendTask(const SessionPacketSendTask&) = delete;
 public:
+    SessionPacketSendTask(const SessionPacketSendTask&) = delete;
```
Comment (ratkosrb) [inline src/game/World.h]: east const

File: src/game/Maps/MapPersistentStateMgr.cpp [hunk 2]
```diff
@@ -491,7 +491,7 @@ void DungeonResetScheduler::ScheduleAllDungeonResets()
     }
 }
 
-void DungeonResetScheduler::ScheduleReset(bool add, time_t time, DungeonResetEvent event)
+void DungeonResetScheduler::ScheduleReset(bool add, time_t const time, DungeonResetEvent event)
```
Comment (ratkosrb) [inline src/game/Maps/MapPersistentStateMgr.cpp]: this is not a pointer so no need for the time parameter to be const

File: src/game/Maps/MapPersistentStateMgr.h [hunk 3]
```diff
@@ -300,7 +300,7 @@ class DungeonResetScheduler
             m_resetTimeByMapId[mapId] = t;
         }
 
-        void ScheduleReset(bool add, time_t time, DungeonResetEvent event);
+        void ScheduleReset(bool add, time_t const time, DungeonResetEvent event);
```
Comment (ratkosrb) [inline src/game/Maps/MapPersistentStateMgr.h]: no need for time to be const

File: src/game/Objects/Creature.cpp [hunk 4]
```diff
@@ -215,7 +215,7 @@ Creature::Creature(CreatureSubtype subtype) :
 
 Creature::~Creature()
 {
-    CleanupsBeforeDelete();
+    Unit::CleanupsBeforeDelete();
```
Comment (ratkosrb) [inline src/game/Objects/Creature.cpp:218]: why explicitly calling unit version

Comment (schell244) [reply src/game/Objects/Creature.cpp:218]: Just for readability -  there's no specific implementation for CleanupsBeforeDelete at creature (while at player there is, which then also calls Unit::CleanupsBeforeDelete).

**Pull request merged**